### PR TITLE
Time varying optimization wip

### PR DESCRIPTION
**Pull request by cliffckerr**
@robynstuart  could you pls do a partial review of this PR, which is running on sandbox, namely:
* https://trello.com/c/SPMrylzR/883-issue-1679-fix-daly-calcs
* https://trello.com/c/6eYMKULu/560-issue-1366-parset-should-contain-start-and-end
* https://trello.com/c/kwYAXRcy/950-fe-wishlist-define-cost-functions-page-programs-past-spending-and-coverage-data-consider-reordering-years-to-be-in-chronological
* https://trello.com/c/lsBgRdSK/900-issue-1696-outcome-mismatch
* https://trello.com/c/9mP1W8DB/946-export-data-for-budget-coverage-scenarios-fails
* https://trello.com/c/NetjgbmJ/947-cascade-plots-are-broken-for-scenarios
* https://trello.com/c/HUunNCxb/959-fe-sandbox-clicking-on-save-run-on-calibration-page-generated-radically-different-results-on-the-care-cascade-graph-from-what-ha

could you please _not_ review:
* time-varying optimization, it's still broken.

other changes: general clean-up, including removing `runmodel()` (yay!).

finally, could you please update the hard-coded indices in `plotting.py` (around line 1000) with the values taken from `project.settings`? i started doing it but got confused so thought it'd be best if you did it.

### CODE REVIEW

File: client/source/js/modules/costfunction/costfunction.js [hunk 1]
```diff
@@ -520,7 +520,7 @@ define(['angular', 'underscore', 'toastr'], function(angular, _) {
           vm.outcomes.push({
             name: vm.state.parameter.short,
             pop: pop,
-            interact: "random",
+            interact: "nested",
```
Comment (robynstuart) [inline client/source/js/modules/costfunction/costfunction.js]: Does this make nested the default?

File: optima/plotting.py [hunk 2]
```diff
@@ -961,6 +996,7 @@ def plotcascade(results=None, aspercentage=False, cascadecolors=None, figsize=gl
             dy = -1
             lineargs = {'c':targetcolor, 'linewidth':2}
             txtargs = {'fontsize':legendsize, 'color':targetcolor, 'horizontalalignment':'center'}
+            print 'WARNING FIX HARD CODING'
```
Comment (robynstuart) [inline optima/plotting.py]: Delete...

File: optima/utils.py [hunk 3]
```diff
@@ -675,9 +783,12 @@ def findinds(val1, val2=None, eps=1e-6):
     return output
 
 
+
 def findnearest(series=None, value=None):
     '''
-    Return the index of the nearest match in series to value
+    Return the index of the nearest match in series to value -- like findinds, but
+    always returns an object with the same time as value (i.e. findnearest with
```
Comment (robynstuart) [inline optima/utils.py]: same type?

File: server/webapp/parse.py [hunk 4]
```diff
@@ -1448,6 +1461,13 @@ def set_optimization_summaries_on_project(project, optimization_summaries):
 
         if "constraints" in summary:
             optim.constraints = revert_constraints(summary['constraints'])
+        
+        try:
+            print('WARNING, TEMP, REMOVE TRY-CATCH')
```
Comment (robynstuart) [inline server/webapp/parse.py]: remove?

File: client/source/js/modules/costfunction/costfunction.js (hunk 1 above)
Comment (cliffckerr) [reply client/source/js/modules/costfunction/costfunction.js]: @robynstuart thx for the eagle eyes, this was a debugging thing i forgot to take out!

File: optima/plotting.py (hunk 2 above)
Comment (cliffckerr) [reply optima/plotting.py]: this is what i want you to fix :)

File: optima/utils.py (hunk 3 above)
Comment (cliffckerr) [reply optima/utils.py]: aye, thx

File: server/webapp/parse.py (hunk 4 above)
Comment (cliffckerr) [reply server/webapp/parse.py]: yes, thx

File: optima/plotting.py (hunk 2 above)
Comment (cliffckerr) [reply optima/plotting.py]: fixed

**Pull request merged**